components/charts/utils/trendCalculator.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Calculate linear regression trend line for a series of data points
+ * @param data - Array of numeric values (must not contain null/NaN values)
+ * @returns Array of trend values with same length as input data
+ */
+export function calculateLinearTrend(data: number[]): number[] {
+  // check if all values are valid numbers
+  if (data.some((value) => isNaN(value) || value === null)) {
+    const invalidCount = data.filter((value) => isNaN(value) || value === null).length;
+    const totalCount = data.length;
+    console.error(
+      `calculateLinearTrend: Data contains ${invalidCount}/${totalCount} invalid values (NaN or null). ` +
+        `Linear regression requires continuous numeric data. ` +
+        `Consider filtering out null values before calling this function. ` +
+        `Returning empty array to prevent chart errors.`
+    );
+    return [];
+  }
```

Comment:
when data is not filtered or validate, it could contain null values, hence adding this check to ensure correct clean up of data before using this function.